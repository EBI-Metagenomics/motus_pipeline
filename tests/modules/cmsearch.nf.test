nextflow_process {

    name "Test module cmsearch"
    script "modules/cmsearch.nf"
    process "CMSEARCH"

    test("cmsearch") {

        when {
            params {
                sequence = "${baseDir}/tests/modules/fixtures/bwa/output/mgtest_merged.fastq.fasta"
                db = "${baseDir}/tests/modules/fixtures/cmsearch/"
            }
            process {
                """
                input[0] = channel.fromPath("${params.sequence}", checkIfExists: true)
                input[1] = channel.fromPath("${params.db}/*.cm", checkIfExists: true).collectFile(name: "models.cm", newLine: true)
                """
            }
        }

        then {
            assert process.success
            assert path(process.out.cmsearch.get(0)).size() == 1103
        }
    }
}