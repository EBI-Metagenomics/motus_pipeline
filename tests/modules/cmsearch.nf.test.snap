{
    "cmsearch": {
        "content": [
            {
                "cmsearch": [
                    "mgtest_merged.fastq.cmsearch_matches.tbl:md5,c68a2c46bc2f30f935c6b1b6f59410e8"
                ]
            }
        ],
        "timestamp": "2023-02-28T14:06:43+0000"
    }
}